### PR TITLE
cleanup

### PR DESCRIPTION
**Pull request by KristofferC**

### CODE REVIEW

File: src/ContMechTensors.jl [hunk 1]
```diff
@@ -214,12 +207,12 @@ for TensorType in (SymmetricTensor, Tensor)
         end
 
         @generated function Base.diagm{order, dim, T}(Tt::Type{$(TensorType){order, dim}}, v::T)
-            if order == 1
```
Comment (fredrikekre) [inline src/ContMechTensors.jl:217]: this will give weird error if you try to do a Vec diagm

Comment (fredrikekre) [reply src/ContMechTensors.jl:217]: "exp not defined"

Comment (KristofferC) [reply src/ContMechTensors.jl:217]: why wouldnt it throw the error i defined below?

Comment (fredrikekre) [reply src/ContMechTensors.jl:217]: Nevermind :)

**Comment by fredrikekre** [comment]
`one` is defined via `diagm` wich now doesnt work for `Vec` :p

**Comment by KristofferC** [comment]
ups, im too boored to fix it today, I'll do it tomorrow unless you get to it :P

File: README.md [hunk 2]
```diff
@@ -2,9 +2,9 @@
 
 *Efficient computations with symmetric and unsymmetric tensors in Julia.*
```
Comment (fredrikekre) [inline README.md]: non-symmetric we call it in the docs.

**Comment by fredrikekre** [comment]
I don't even know if we should keep the same `one` behaviour as previously; with `one` being for `Vec`s a `Vec` filled with ones and for second and fourth order basically the unit tensor. It is not consistent IMO.
Maybe we should instead define `eye` for second and fourth order, and completely remove `one`?

**Comment by fredrikekre** [comment]
From docs:

``` jl
help?> one
search: one ones done spones exponent convert connect nonzeros istaskdone

  one(x)

  Get the multiplicative identity element for the type of x (x can also
  specify the type itself). For matrices, returns an identity matrix of the
  appropriate size and type.
```

so based on this

``` jl
one(::Vec) # should return second order unit tensor
one(::Tensor{2}) # should return fourth order unit tensor
one(::Tensor{4}) # should return 6? order unit tensor
```

But its a bit weird since we don’t define `*` but rather different types of contractions.

**Comment by KristofferC** [comment]
The multiplicative identity for `Tensor{2}` has to be an identity second order tensor, no?

**Comment by fredrikekre** [comment]
Yea right, so multiplicative identity would be using `\cdot` for `Vec` and `Tensor{2}` then.

**Comment by KristofferC** [comment]
I guess `one` for `Vec` shouldn't be defined. We should define `ones` instead for all tensors which would just be all ones.

**Comment by fredrikekre** [comment]
I mean, we could have this:

``` jl
julia> V = rand(Vec{3}); T2 = rand(Tensor{2,3}); T4 = rand(Tensor{4,3});

julia> one(V) ⋅ V == V
true

julia> one(T2) ⋅ T2 == T2
true

julia> one(T4) ⊡ T4 == T4
true
```

which makes sense to me. +1 for adding a ones, could be used to create a tensor with constants then too, like

``` jl
julia> 2 * ones(Tensor{2,3})
3×3 ContMechTensors.Tensor{2,3,Float64,9}:
 2.0  2.0  2.0
 2.0  2.0  2.0
 2.0  2.0  2.0
```

**Comment by KristofferC** [comment]
I think it is weird if `one(::T)` is not of type `T`.

**Comment by fredrikekre** [comment]
But doesn’t work with `\cdot` then. If we wanna use `\cdot` as the base multiplication of tensors. But its ok to be undefined I guess.

File: test/runtests.jl [hunk 3]
```diff
@@ -21,7 +21,7 @@ for T in (Float32, Float64)
             # Simple math #
             ###############
             t = rand(Tensor{order, dim, T})
-            t_one = one(Tensor{order, dim, T})
+            t_ones = ones(Tensor{order, dim, T})
```
Comment (fredrikekre) [inline test/runtests.jl:24]: This is not even used in the tests.

**Comment by fredrikekre** [comment]
Documented ones too, review and merge!

File: docs/src/man/constructing_tensors.md [hunk 4]
```diff
@@ -43,6 +43,26 @@ julia> zero(SymmetricTensor{4, 2, Float32})
  0.0  0.0
 ```
 
+## Constant tensors
```
Comment (fredrikekre) [inline docs/src/man/constructing_tensors.md:46]: Idk about this header. `Uniform tensors`? `One tensors`?

File: test/runtests.jl (hunk 3 above)
Comment (KristofferC) [reply test/runtests.jl:24]: Kill it then :P

Comment (fredrikekre) [reply test/runtests.jl:24]: But it should probably be used or tested somehow :p
Can rewrite constructor tests in a separate PR

File: docs/src/man/constructing_tensors.md (hunk 4 above)
Comment (KristofferC) [reply docs/src/man/constructing_tensors.md:46]: Ok by me to keep.

**Comment by KristofferC** [comment]
Thx!

**Pull request merged**